CIP-Thompson-AuditBestPractice/CIP-XXX.md

```diff
@@ -0,0 +1,209 @@
+
+*Note*: “certification” has also been used informally to cover the combined process of testing, audit, verification and providing evidence of these, as in “level 1 certification”. If it is felt that it is too confusing to use the term in both senses, then another term should be found, e.g. *evidencing* or *witnessing*. 
+
+***Component***: a discrete part of the complete system supplied by a third party, such as a wallet, (part of) which will be run on the user’s system.
+
+***Service***: a part of the system, such as Blockfrost, that is provided by an online server accessed through a well-defined protocol.
+
+***Scope***: the parts of the DApp that are subject to audit. A repository may contain much more than Plutus code. Some audits will look at the complete app, some will just concentrate on on-chain transactions, others on the web interface around it. The possible scopes are divided in three main categories:
+* On-chain
+* Off-chain
+* Context: the business and economic models for the DApp  
+
+***Deployment***: once a DApp is developed it is deployed by submitting the relevant on-chain scripts onto the Cardano blockchain as transaction validator scripts. The scripts being deployed  might be the scripts that have been audited, or be instances of them in the case that the audited scripts are *parametric*.
+
+###  Tendering
+In order to provide a quote for audit, developers will need to supply
+* A specification of the DApp to be audited (more details below).
+* The scope of the audit.
+* An estimate of the scale of the audit work, e.g. the number of lines in the on-chain code to be audited, or the code itself, in its current state of development.
+
+### Submission
+In order to be audited, developers will need to supply the following documentation.
+
+#### *Specification / design documents*
+
+Submitters shall provide specification and design documents that describe in a precise and unambiguous way the architecture, interfaces and requirements characterising the DApp. 
+
+The documentation shall identify the expected behaviour of the code, given without direct reference to the code itself. The description should also include high-level examples demonstrating use cases of the DApp. All assumptions about how the DApp will be used will be described. The documentation shall identify and document all the interfaces with other components and services.
+
+Submitters might also wish to explain mitigating actions that they have taken to protect against potential failures and attacks to the DApp.
+
+#### *On-Chain Specification* 
+
+The format of transactions accepted by the smart contracts should be specified using the template [here](https://docs.google.com/document/d/1j1s17Sdb7i8NzhK5GvgvcZqYbBtUzJB9Qpalmk8xElY). 
```

Comment:
May we make that document a markdown document attached to this particular CIP (you may put multiple files in the CIP folder).

Reply:
Certainly we could do that. We could also make it a separate CIP. I'll do the former for the moment.

Reply:
Done.